datafusion/execution/src/cache/cache_manager.rs

```diff
@@ -102,12 +118,19 @@ impl CacheManager {
     }
 
     /// Get the file embedded metadata cache.
-    pub fn get_file_metadata_cache(&self) -> Option<Arc<dyn FileMetadataCache>> {
-        self.file_metadata_cache.clone()
+    pub fn get_file_metadata_cache(&self) -> Arc<dyn FileMetadataCache> {
+        Arc::clone(&self.file_metadata_cache)
+    }
+
+    /// Get the limit of the file embedded metadata cache.
+    pub fn get_file_metadata_cache_limit(&self) -> Option<usize> {
+        self.file_metadata_cache.cache_limit()
     }
 }
 
-#[derive(Clone, Default)]
+const DEFAULT_FILE_METADATA_CACHE_LIMIT: usize = 1024 * 1024 * 1024; // 1G
```

Comment:
The default limit is set to `1G`. I'm not sure if this is an appropriate default.

Reply:
Agreed I think it should be pretty conservative (small)

Reply:
Changed to 50MB.

Reply:
I checked on a 100M row and 2 col file, with 1.5k pages, and it had ~5MB of metadata, so the 50MB limit seems appropriate.